src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -894,6 +920,14 @@ private boolean handleJmxNotificationForCassandra21(
                     .withId(segmentId)
                     .build());
 
+            // Since we can get out of order notifications,
+            // we need to exit if we already got the COMPLETE notification.
+            gotSuccessNotification.set(Boolean.TRUE);
+            if (gotCompleteNotification.get()) {
```

Comment:
this will never be true

Reply:
This is precisely what we're trying to deal with. We get cases where `COMPLETE` gets received before `SUCCESS` does.
What makes you think it can never be true ?